aws/src/integration/java/org/apache/iceberg/aws/AwsIntegTestUtil.java

```diff
@@ -127,6 +129,21 @@ public static void cleanS3Bucket(S3Client s3, String bucketName, String prefix)
     }
   }
 
+  /**
+   * Method used to clean up a S3 directory bucket which doesn't care about versions
+   *
+   * @param s3FileIO an instance of s3FileIO to be used to list/delete objects
+   * @param bucketName name of the bucket
+   * @param prefix the path prefix we want to remove
+   */
+  public static void cleanS3DirectoryBucket(S3FileIO s3FileIO, String prefix, String bucketName) {
```

Comment:
I general I think we should support directory bucket listing, but we should not use that to clean test bucket. The reason `cleanS3GeneralPurposeBucket` did not use S3FileIO is that this is the test suite for S3FileIO, if the logic is somehow wrong then the listing and cleanup would break. We want to directly call the raw client to perform the delete.

Reply:
So it was like like that in the first revision, changed based on recommendations from the prior reviews.
I guess if S3FileIO.deleteFiles does go wrong it would be picked up by the tests doing the exact same thing here.

Reply:
yes, that's why I think it is important to not use those calls in test cleanup logic